# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/JitFrames.cpp
# Commit: c012fb62e4fe
# Full Hash: c012fb62e4fe45d3c0ec02bd9fb26bb94b3da9fb
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2022-06-09 22:00:48
# Regressor Bug: 1770366
# File Overlap Count: 2
# Description:
#   Bug 1770366 part 5 - Add frame pointer to IonJS/Warp frames. r=iain
#   
#   Change OSR to use the standard frame epilogue on the Baseline side and frame prologue
#   on the CodeGenerator side.
#   
# ==============================================================================

diff -r cc3833b385cd -r c012fb62e4fe js/src/jit/JitFrames.cpp
--- a/js/src/jit/JitFrames.cpp	Thu Jun 09 13:49:16 2022 +0000
+++ b/js/src/jit/JitFrames.cpp	Thu Jun 09 13:49:17 2022 +0000
@@ -862,7 +862,7 @@
 
 uintptr_t* JitFrameLayout::slotRef(SafepointSlotEntry where) {
   if (where.stack) {
-    return (uintptr_t*)((uint8_t*)this - where.slot);
+    return (uintptr_t*)((uint8_t*)this - IonFirstSlotOffset - where.slot);
   }
   return (uintptr_t*)((uint8_t*)argv() + where.slot);
 }